### ISSUE TITLE
getSymmetryOrder(AFPChain) gives unexpected results

### ISSUE THREAD

**Issue by dmyersturnbull** [biojava/biojava#88]
This first snippet seems to return 1:
```
int order = getSymmetryOrder(afpChain, 8, 0.4f);
```
This snipped does what I expect:
```
Map<Integer,Integer> alignment = AlignmentTools.alignmentAsMap(afpChain);
int order = AlignmentTools.getSymmetryOrder(alignment, new AlignmentTools.IdentityMap<Integer>(), 8, 0.4f);
```

What is the difference between these two overloaded methods?

**Comment by lafita** [comment]
Could you @sbliven have a look at this?

**Comment by sbliven** [comment]
I looked into this.

I can in fact reproduce the bug with highly-symmetric cases where you would expect the SequenceFunctionOrderDetector to work well:
```
	public static void main(String[] args) throws IOException, StructureException {
		String name = "1itb.A";
		Structure s = StructureIO.getStructure(name);
		Atom[] ca = StructureTools.getRepresentativeAtomArray(s);
		CeSymmResult result = CeSymm.analyze(ca);
		AFPChain afp = result.getSelfAlignment();
		
		int order1 = AlignmentTools.getSymmetryOrder(afp, 8, .4f);
		Map<Integer,Integer> alignment = AlignmentTools.alignmentAsMap(afp);
		int order2 = AlignmentTools.getSymmetryOrder(alignment, new AlignmentTools.IdentityMap<Integer>(), 8, 0.4f);
		System.out.format("order1=%d%norder2=%d%n",order1,order2);
	}
```
(Yields `order1=1 order2=3`)

These methods are part of SymmetryTools, but they were actually written without assuming that the rows of the alignment correspond to the same protein. This was useful when looking at the symmetry of CE-CP alignments, although now I wonder if those results can actually be trusted. Anyways, the methods require an _identity alignment_ which gives a sequential alignment between the residues in each structure of the multiple alignment. For internal symmetry, where the structures are truly identical, an IdentityMap should be used. Thus, the second method is the correct way to calculate the order for internal symmetry. Unfortunately, if no identity alignment is manually specified, it uses the very naive `guessSequentialAlignment` instead. This method assumes that the input alignment is refined (or nearly so). This makes it quite sensitive to gaps in coverage and to register shifts in the self alignment, and it seems to give bad results even in highly-symmetric cases like 1itb.A (beta-trefoil).

With a refined alignment, `guessSequentialAlignment` is equivalent to the IdentityMap, so the method works as expected (with some ugly adaptor code to fit CeSymmResults into the old AFPChain-expecting framework):

```
		MultipleAlignment ma = SymmetryTools.toFullAlignment(result);
		List<Atom[]> atoms = ma.getAtomArrays();
		alignment.clear();
		for(Block block : ma.getBlocks()) {
			List<List<Integer>> aln = block.getAlignRes();
			for(int i = 0; i<block.length();i++) {
				alignment.put(aln.get(0).get(i),aln.get(1).get(i));
			}
		}
		afp = replaceOptAln(afp, atoms.get(0), atoms.get(1), alignment);
		int order3 = AlignmentTools.getSymmetryOrder(afp, 8, .4f);
```

TLDR: This is expected. I'll add a note to the documentation and close this issue.
